tests/testthat/test-helpers_knitr_Stopping.R

```diff
@@ -0,0 +1,413 @@
+      "range [0.20, 0.35] is at least 0.50.\n-  If 20 or more participants ",
+      "have been treated."
+    )
+  )
+  x@report_label <- NA_character_
+  expect_equal(
+    knit_print(x, asis = FALSE),
+    paste0(
+      "If any of the ",
+      "following rules are `TRUE`:\n\n-  If 3 or more cohorts have been treated.",
+      "\n-  If the probability of toxicity at the next best dose is in the ",
+      "range [0.20, 0.35] is at least 0.50.\n-  If 20 or more participants ",
+      "have been treated."
+    )
+  )
+})
+
+test_that("knit_print.StoppingList works correctly", {
+  x <- .DefaultStoppingList()
+  x@report_label <- "LIST"
+  expect_equal(
+    knit_print(x, asis = FALSE),
+    paste0(
+      "LIST: If the result of applying the summary function to the ",
+      "following rules is `TRUE`:\n\n-  If 3 or more cohorts have been treated.",
+      "\n-  If the probability of toxicity at the next best dose is in the ",
+      "range [0.20, 0.35] is at least 0.50.\n-  If 20 or more participants ",
+      "have been treated."
+    )
+  )
+  x@report_label <- NA_character_
+  expect_equal(
+    knit_print(x, asis = FALSE),
+    paste0(
+      "If the result of applying the summary function to the ",
+      "following rules is `TRUE`:\n\n-  If 3 or more cohorts have been treated.",
+      "\n-  If the probability of toxicity at the next best dose is in the ",
+      "range [0.20, 0.35] is at least 0.50.\n-  If 20 or more participants ",
+      "have been treated."
+    )
+  )
+})
+
+test_that("knit_print.StoppingMaxGainCIRatio works correctly", {
+  expect_equal(
+    knit_print(
+      StoppingMaxGainCIRatio(target_ratio = 4, prob_target = 0.3, report_label = "MAXGAINCI_RATIO"),
+      asis = FALSE
+    ),
+    paste0(
+      "MAXGAINCI_RATIO: If the ratio of the upper and lower limits of the ",
+      "posterior 95% credible interval for the probability of toxicity at the ",
+      "smaller of the doses given by the end-of-trial TD estimate and the G* ",
+      "estimate is less than or equal to 4."
+    )
+  )
+  expect_equal(
+    knit_print(
+      StoppingMaxGainCIRatio(target_ratio = 4, prob_target = 0.3),
+      asis = FALSE,
+      dose_label = "the MTD"
+    ),
+    paste0(
+      "If the ratio of the upper and lower limits of the ",
+      "posterior 95% credible interval for the probability of toxicity at the ",
+      "smaller of the doses given by the end-of-trial TD estimate and the G* ",
+      "estimate is less than or equal to 4."
+    )
+  )
+})
+
+test_that("knit_print.StoppingTDCIRatio works correctly", {
+  expect_equal(
+    knit_print(
+      StoppingTDCIRatio(target_ratio = 4, prob_target = 0.3, report_label = "TDCI_RATIO"),
+      asis = FALSE
+    ),
+    paste0(
+      "TDCI_RATIO: If, at the next best dose, the ratio of the upper and lower ",
+      "limits of the posterior 95% credible interval for the probability of ",
+      "toxicity is less than or equal to 4."
```

Comment:
```suggestion
      "TDCI_RATIO: If, at the next best dose, the ratio of the upper to the lower ",
      "limit of the posterior 95% credible interval for the MTD (targeting 30%) ",
      "is less than or equal to 4."
```

Reply:
would "... (targeting p(Tox) = 30%) ..." be better?  (And similarly for `StoppingMaxGainCIRatio`...)

Reply:
hm not sure, I would not put too much math notation in there